include/seqan3/core/algorithm/detail/execution_handler_sequential.hpp

```diff
@@ -12,40 +12,36 @@
 
 #pragma once
 
-#include <functional>
+#include <seqan3/std/concepts>
 
-#include <seqan3/alignment/pairwise/detail/concept.hpp>
 #include <seqan3/core/platform.hpp>
-#include <seqan3/range/views/type_reduce.hpp>
-#include <seqan3/std/concepts>
-#include <seqan3/std/ranges>
 
 namespace seqan3::detail
 {
 
-/*!\brief Handles the sequential execution of alignments.
- * \ingroup execution
+/*!\brief Handles the sequential execution of algorithms.
+ * \ingroup algorithm
  */
 struct execution_handler_sequential
 {
 public:
 
-    /*!\brief Takes underlying range of sequence pairs and invokes an alignment on each instance.
-     * \tparam algorithm_t              The type of the alignment algorithm.
-     * \tparam indexed_sequence_pairs_t The type of underlying sequence pairs annotated with an index.
-     * \tparam delegate_type            The type of the callable invoked on the std::invoke_result of `algorithm_t`.
+    /*!\brief Executes the algorithm with the given input and callback.
+     * \tparam algorithm_t The type of the algorithm.
+     * \tparam algorithm_input_t The input type to invoke the algorithm with.
+     * \tparam callback_t The type of the callable invoked by the algorithm after generating a new result.
      *
-     * \param[in] algorithm              The alignment algorithm to invoke.
-     * \param[in] indexed_sequence_pairs The range of underlying annotated sequence pairs to be aligned.
-     * \param[in] delegate               A callable which will be invoked on each result of the computed alignments.
+     * \param[in] algorithm The algorithm to invoke.
+     * \param[in] input The input of the algorithm.
+     * \param[in] callback A callable which will be invoked on each result generated by the algorithm.
      */
-    template <typename algorithm_t, typename indexed_sequence_pairs_t, typename delegate_type>
-    void execute(algorithm_t && algorithm,
-                 indexed_sequence_pairs_t && indexed_sequence_pairs,
-                 delegate_type && delegate)
+    template <typename algorithm_t, typename algorithm_input_t, typename callback_t>
+    //!\cond
+        requires std::invocable<algorithm_t, algorithm_input_t, callback_t>
+    //!\endcond
+    void execute(algorithm_t && algorithm, algorithm_input_t && input, callback_t && callback)
     {
-        algorithm(std::forward<indexed_sequence_pairs_t>(indexed_sequence_pairs),
-                  std::forward<delegate_type>(delegate));
+        algorithm(std::forward<algorithm_input_t>(input), std::forward<callback_t>(callback));
     }
 
     //!\brief Waits for the submitted alignments jobs to finish. (Noop).
```

Comment:
here is still the `alignment` keyword. Should be changed to something like:
waits for the sequentially executed algorithm to finish.